### PR TITLE
WIP: Notes: Custom order

### PR DESCRIPTION
**Pull request by OlegMoshkovich**
This PR is WIP - it addresses #311 and #319.
Added note# parameter to the note to sort the notes according to custom order number.

the link to test the notes order is here:
https://deploy-preview-352--bldrs-share.netlify.app/share/v/gh/OlegMoshkovich/BLDRS_models/main/haus.ifc/66/389/434/35065/59753

test issues are defined here:
https://github.com/OlegMoshkovich/Sample_project/issues

### CODE REVIEW

**Review by OlegMoshkovich — COMMENTED** [review]
Some comments for the utility functions
This is still work in progress, but the structure is emerging.
@pablo-mayrgundter if you have time, please take a look and comment.

File: src/utils/objects.js [hunk 1]
```diff
@@ -20,3 +20,17 @@ export function isObject(obj) {
...
   return obj === Object(obj) && Object.prototype.toString.call(obj) !== '[object Array]'
 }
+
+
+/**
+ * @param {Array} array
+ * @param {String} key
+ * @return {Array} array sorted according to the key
+ */
+export function sortObjectsByKey(array, key) {
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/utils/objects.js:30]: Cool.  Test pls.

File: src/utils/strings.js [hunk 2]
```diff
@@ -42,3 +42,32 @@ export function findUrls(str) {
...
   })
 }
+
+
+/**
+ * @param {string} str
+ * @return {string} content of the body
+ */
+export function notesOrder(str) {
+  let extractedNumber
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/utils/strings.js:52]: I think for this we should find or write a markdown plugin.   Check out the docs here: https://www.npmjs.com/package/react-markdown#api

File: src/utils/strings.js [hunk 3]
```diff
@@ -42,3 +42,32 @@ export function findUrls(str) {
...
+    return 0
+  }
+}
+
+
+/**
+ * @param {string} str
+ * @return {string} content of the body
+ */
+export function extractBodyContent(str) {
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/utils/strings.js:66]: Neato :) Not sure if this is better plugin or not.  Can you try doing extract above as a plugin and then let's see?

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

File: src/utils/strings.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (OlegMoshkovich) [reply src/utils/strings.js:68]: how do you mean plugin?

**Comment by OlegMoshkovich** [comment]
@pablo-mayrgundter 
This PR is blocking me.
I cannot find a plug-in.
Seems like a more involved effort.
closing it.

**Pull request closed without merging**